api/v1alpha5/conversion.go

```diff
@@ -220,6 +220,8 @@ func Convert_v1alpha5_OpenStackMachineSpec_To_v1alpha8_OpenStackMachineSpec(in *
 
 	if in.ServerGroupID != "" {
 		out.ServerGroup = &infrav1.ServerGroupFilter{ID: in.ServerGroupID}
+	} else {
+		out.ServerGroup = &infrav1.ServerGroupFilter{}
```

Comment:
Hm, do we want to set it if ServerGroupID was empty? Feels counter-intuitive to me, but I'm pretty new to all these conversion things.

Reply:
it's the default, since ServerGroup is not a pointer (otherwise default would have been nil). If you remove the else, the output will be ServerGroup: nil. Try it!

Reply:
Meh I read-proof myself and I'm wrong this is a pointer. However I believe that's because a pointer that we need to define it.